metrics/qos/evm/metrics.go

```diff
@@ -49,13 +58,92 @@ var (
 	//   - End-to-end request success rates
 	//   - Error types by JSON-RPC method and chain
 	//   - HTTP status code distribution
+	//   - Service degradation when random endpoint fallback is used
 	requestsTotal = prometheus.NewCounterVec(
 		prometheus.CounterOpts{
 			Subsystem: pathProcess,
 			Name:      requestsTotalMetric,
 			Help:      "Total number of requests processed by EVM QoS instance(s)",
 		},
-		[]string{"chain_id", "service_id", "request_origin", "request_method", "success", "error_type", "http_status_code"},
+		[]string{"chain_id", "service_id", "request_origin", "request_method", "success", "error_type", "http_status_code", "random_endpoint_fallback"},
+	)
+
+	// availableEndpoints tracks the number of available endpoints per service.
+	// Labels:
+	//   - chain_id: Target EVM chain identifier
+	//   - service_id: Service ID of the EVM QoS instance
+	//
+	// Use to analyze:
+	//   - Endpoint pool size per service
+	//   - Service capacity trends
+	availableEndpoints = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Subsystem: pathProcess,
+			Name:      availableEndpointsMetric,
+			Help:      "Number of available endpoints for EVM QoS instance(s)",
+		},
+		[]string{"chain_id", "service_id"},
```

Comment:
1. Can you please add the following:
- `session-id`
- `block-height`

2. I'm also thinking if we should be tracking the number of available nedpoints OR the number of endpoints with unique TLDs.

Consider adding a TODO.